ApplicationLibCode/ProjectDataModel/Summary/RimSummaryRegressionAnalysisCurve.cpp

```diff
@@ -110,11 +117,54 @@ RimSummaryRegressionAnalysisCurve::~RimSummaryRegressionAnalysisCurve()
 //--------------------------------------------------------------------------------------------------
 void RimSummaryRegressionAnalysisCurve::onLoadDataAndUpdate( bool updateParentPlot )
 {
+    auto xValues    = RimSummaryCurve::valuesX();
+    auto yValues    = RimSummaryCurve::valuesY();
+    auto timeStepsX = RimSummaryCurve::timeStepsX();
+    auto timeStepsY = RimSummaryCurve::timeStepsY();
+
+    if ( axisTypeX() == RiaDefines::HorizontalAxisType::SUMMARY_VECTOR )
+    {
+        // NB! Assume that time stamps for X and Y are the same
+        std::vector<size_t> indicesToRemove;
+
+        // Step 1: Find indices of values which are outside the specified range
+        for ( size_t i = 0; i < xValues.size(); i++ )
+        {
+            if ( xValues[i] < m_valueRangeX().first || xValues[i] > m_valueRangeX().second )
+            {
+                indicesToRemove.push_back( i );
+            }
+        }
+
+        for ( size_t i = 0; i < yValues.size(); i++ )
+        {
+            if ( yValues[i] < m_valueRangeY().first || yValues[i] > m_valueRangeY().second )
+            {
+                indicesToRemove.push_back( i );
+            }
+        }
+
+        // Step 2: Sort indices in descending order
+        std::sort( indicesToRemove.rbegin(), indicesToRemove.rend() );
```

Comment:
Also unique the results? It could be that the same index is remove twice if it matches both x and y criteria?

Reply:
Fixed, good catch :-)